### ISSUE TITLE
o.e.j.servlet.listener.ContainerInitializer does not respect servlet spec contract for ServletContainerInitializers

### ISSUE THREAD

**Issue by janbartel** [eclipse/jetty.project#5834]
**jetty 9.4.x**

See https://github.com/eclipse/jetty.project/pull/3706.

The premise of the org.eclipse.jetty.servlet.listener.ContainerInitializer according to the comments is to ease using ServletContainerInitializers in embedded scenarios. Unfortunately, it calls the SCI from a ServletContextListener: according to the spec, a ServletContextListener cannot add another ServletContextListener. Thus, an SCI called via the ContainerInitializer will not be able to add a ServletContextListener, as it is perfectly entitled to do.

**Comment by gregw** [comment]
One solution may be to use a jetty lifecycle listener instead of a ServletContextListener

**Comment by joakime** [comment]
for embedded-jetty, I would love to see an API like ...

``` java
ServletContextHandler context = new ServletContextHandler();
context.addServletContextInitializer(new MySCI());
// other init
context.start();
```

Then let Jetty work out the internals of how that should come to be.

**Comment by gregw** [comment]
@lachlan-roberts I think you are already addressing this one in #5840 
I think you are just deprecating it rather than fixing it... @janbartel is that OK?

**Comment by janbartel** [comment]
I'm ok with deprecation if it is entirely removed in jetty-10 and above.

**Comment by lachlan-roberts** [comment]
`ContainerInitializer.ServletContainerInitializerServletContextListener` is a public class so we cant just remove it. And we have already done a release of jetty-10, so I think the best thing we can do is deprecate it right now.

**Comment by lachlan-roberts** [comment]
This has been fixed with PR #5959.
